third_party/move/move-vm/runtime/src/storage/ty_layout_converter.rs

```diff
@@ -103,6 +106,120 @@ where
+                                }
+                            }
+                        },
+                        StructLayout::Variants(variants) => {
+                            for variant in variants {
+                                for (_, field_ty) in &variant.1 {
+                                    if self.contains_option_type(
+                                        field_ty,
+                                        gas_meter,
+                                        traversal_context,
+                                    )? {
+                                        return Ok(true);
+                                    }
+                                }
+                            }
+                        },
+                    }
+                    Ok(false)
+                }
+            },
+            StructInstantiation { idx, ty_args, .. } => {
+                let struct_identifier = self
+                    .runtime_environment()
+                    .struct_name_index_map()
+                    .idx_to_struct_name(*idx)?;
+                if struct_identifier.module == option_module_id
+                    && struct_identifier.name == option_struct_name
+                {
+                    Ok(true)
+                } else {
+                    let definition = self.struct_definition_loader.load_struct_definition(
+                        gas_meter,
+                        traversal_context,
+                        idx,
+                    )?;
+                    match &definition.layout {
+                        StructLayout::Single(fields) => {
+                            let tys = self.apply_subst_for_field_tys(fields, ty_args)?;
+                            for ty in tys.iter() {
+                                if self.contains_option_type(ty, gas_meter, traversal_context)? {
+                                    return Ok(true);
+                                }
+                            }
+                        },
+                        StructLayout::Variants(variants) => {
+                            for variant in variants {
+                                let tys = self.apply_subst_for_field_tys(&variant.1, ty_args)?;
+                                for ty in tys.iter() {
+                                    if self.contains_option_type(
+                                        ty,
+                                        gas_meter,
+                                        traversal_context,
+                                    )? {
+                                        return Ok(true);
+                                    }
+                                }
+                            }
+                        },
+                    }
+                    Ok(false)
+                }
+            },
+            // No need to check results because they will not appear in the layout of captured values
+            Function { args, .. } => {
```

Comment:
This is not correct? args also do not appear, because captured arguments are not visible in the type

Reply:
so we don't actually need this branch?